### PR TITLE
Remove `surface` parameter

### PR DESCRIPTION
**Pull request by garyb**
An option discussed in #613

### CODE REVIEW

File: examples/higher-order-components/src/Harness.purs [hunk 1]
```diff
@@ -26,7 +26,7 @@ type ChildSlots =
 _panel :: SProxy "panel"
 _panel = SProxy
 
-component :: forall f i o m. H.Component HH.HTML f i o m
+component :: forall f i o m. H.Component f i o m
```
[the remark this answers is not in the document]
Comment (garyb) [reply examples/higher-order-components/src/Harness.purs]: Yeah, definitely 👍 . `f` is left over from before we gave the parameters long names. I did consider using long names in all these examples too, but it's kinda the point that the types don't matter when components are instantiated this way, so I didn't want to give them suggestive names.

**Comment by thomashoneyman** [comment]
I've had some more time to think about this, and in that time I've leaned even more towards removing `surface` rather than making it even more present (the associated types PR). I think this PR is the one to merge, and if / when `surface` is desired in the future it can be re-introduced.

**Comment by garyb** [comment]
I agree, just gotta figure out how to juggle the release stuff for it now. It's kinda similar to the event-rather-than-`EventSource` PR, in that it's a breaking change for the current material that's out there for H5.

**Comment by thomashoneyman** [comment]
Perhaps there could be a Halogen 6 branch, where these various PRs can be merged? I could work from that branch to add support for portals, too.

**Comment by garyb** [comment]
Yeah, good thinking. I'll get that going 🙂.

**Pull request merged**